nmigen/hdl/ast.py

```diff
@@ -1186,7 +1186,10 @@ def lower(self):
 
     def _lazy_lower(self):
         if self.__lowered is None:
-            self.__lowered = Value.cast(self.lower())
+            lowered = self.lower()
+            if isinstance(lowered, UserValue):
+                lowered = lowered._lazy_lower()
```

Comment:
Could you please write a test for lowering of a `UserValue` that returns another `UserValue`?